fastlane/Fastfile

```diff
@@ -142,9 +142,23 @@ platform :android do
     trigger_release_build(branch_to_build: "release/#{new_version}")
   end
 
+  # @option [String] base_version The "x.y" version number to create a beta from. Defaults to the current version as defined in the version file of `main` branch
+  # @option [Integer] version_code The versionCode to use for the new beta. Defaults to one more than the versionCode defined in the version file of the `release/<base_version>` branch
   lane :new_beta_release do |options|
     ensure_git_status_clean
-    Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
+
+    base_version = options[:base_version]
+    if base_version.nil?
+      # If no base_version provided, read it from the version file from `main`
+      Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
+      base_version = release_version_current
+    end
+
+    # Checkout release branch first, so that all computations of `beta_version_*`` and `build_code_*`` are based on the
+    # current values in the current `release/*`` branch, not the values from `main` (which is important in cases like
+    # hotfixes for example, where `main` doesn't have the new versionCode used by the hotfix yet but release/* branch do)
+    checkout_success = Fastlane::Helper::GitHelper.checkout_and_pull(release: base_version)
+    UI.user_error!("Release branch for version #{base_version} doesn't exist.") unless checkout_success
```

Comment:
ℹ️  Note that PCiOS [uses a different approach, where there's no code trying to be clever in guessing which version you want to do a beta for and which branch to check out](https://github.com/Automattic/pocket-casts-ios/blob/trunk/fastlane/Fastfile#L486-L487).
Instead, in PCiOS it's expected that the release manager would check out the right `release/*` branch manually before calling the `new_beta_release` lane from the `release/*` branch.

This is also how we do it in Tumblr iOS & Tumblr Android, btw.

Once we'll migrate PocketCasts to Release-on-CI (i.e. make all the lanes be run by CI, including letting CI call the `new_beta_release` lane itself, to avoid requiring release managers to have to run the commands from their local Terminal and have Ruby installed and configured on their machine to do so), we'll be able to tell the button in MC's ReleasesV2 scenario to trigger the `new_beta_release` CI job directly from the right `release/%version%` branch for the `x.y` version of that scenario.

So assuming that we're already on the right branch (like PCiOS does) will probably be the right approach in the long run—instead of trying to be clever guessing the release branch (like it's done here)—to be ready for when that lane will also run in CI. But doing so right now would also require to update the instructions in PCAndroid's release scenario in MC to explicitly tell the RM to checkout the release branch first, etc.

So given this PR was more reactive, to fix a bug, than as part of migrating PCAndroid to release-in-CI, I figured that this PR should only fix the prompt bug but still keep the existing behavior of the rest (i.e. guessing the release branch to checkout) for now, and we can take some more time to align PCiOS and PCAndroid + update the release scenarios in MC at a later time (when someone has the bandwidth or when we start implementing release-on-CI for PocketCasts at some point)